src/ch01/array_copy_caf.f90

```diff
@@ -18,7 +18,7 @@ program array_copy_caf
 
 sync all
 
-array(:)[2] = array(:)[1]
+if (this_image() == 2) array(:) = array(:)[1]
```

Comment:
If you want you could even drop the if statement. Two copies would still happen but only one would be involving a remote coarray. (i.e. image 1 would just do `array(:) = array(:)[1]` which is equivalent to `array = array` and image 2 would do `array = array[1]` performing a GET.

FYI, shape, allocation status, and type must match when remote coarrays are involved, so the `(:)` is redundant---any sort of reallocation on assignment of the LHS would be a compiler bug and violate the standard. I will probably eliminate that in a commit in a second.